app-shells/command_not_found/patches/command_not_found-0.0.1~git.patchset

```diff
@@ -0,0 +1,84 @@
+From 192f2f912f75b7304f09f2b255e24427eef800c1 Mon Sep 17 00:00:00 2001
+From: tts2k <tranthaison2000@gmail.com>
+Date: Sun, 7 Jan 2018 02:49:19 +0000
+Subject: Add read permission
+
+
+diff --git a/haiku_cnf.py b/haiku_cnf.py
+index 7c68057..c7f51d9 100644
+--- a/haiku_cnf.py
++++ b/haiku_cnf.py
+@@ -17,7 +17,7 @@ def update_db():
+     db['base-pkgs'] = json.dumps(read_basepkgs())
+     if Popen(['which', 'haikuporter'],stdout=PIPE).communicate()[0]:
+         home = os.path.expanduser('~')
+-        with open("%s/config/settings/command-not-found/options.json" % home, "w") as options: 
++        with open("%s/config/settings/command-not-found/options.json" % home, "rw") as options: 
+             d = json.load(options)
+             d['haikuports'] = True
+             # json.dump(d, options)
+diff --git a/install_cnf.py b/install_cnf.py
+index 1e1dd35..fc0051b 100644
+--- a/install_cnf.py
++++ b/install_cnf.py
+@@ -35,7 +35,6 @@ def main():
+         options = open("%s/config/settings/command-not-found/options.json" % home, "w")
+         options.write(default_options)
+         options.close()
+-    os.system("install -m 755 haiku_cnf.py /boot/system/non-packaged/bin/command_not_found.py")
+ 
+ 
+ if __name__ == '__main__':
+-- 
+2.15.0
+
+
+From c22e9f9d3f050a2d3f156314b7b9e3ec87df2731 Mon Sep 17 00:00:00 2001
+From: tts2k <tranthaison2000@gmail.com>
+Date: Mon, 8 Jan 2018 01:03:46 +0000
+Subject: Fix post-install script
+
+
+diff --git a/install_cnf.py b/install_cnf.py
+index fc0051b..e65ceb7 100644
+--- a/install_cnf.py
++++ b/install_cnf.py
+@@ -1,10 +1,11 @@
+-#!/bin/env python
++#!/bin/python
+ import os
++import commands
+ 
+ hacks = """# command-not-found tomfoolery
+ if [ -e /system/bin/python ]; then
+     command_not_found_handle(){
+-        /boot/system/bin/python /boot/system/non-packaged/bin/command_not_found.py "$1"
++        /boot/system/bin/python /boot/system/bin/command_not_found "$1"
```

Comment:
`finddir B_SYSTEM_BIN_DIRECTORY` currently returns `/boot/system/bin`, but this could change some day, in _theory_.
Hence this question: should we replace `/boot/system/bin` by the result of the call to finddir?
Or is this a bad idea?
I really don't know.
If we keep this hardcoded path, then we will need to update the patchset if we ever change `B_SYSTEM_BIN_DIRECTORY`.
On the other side, I guess such a change is unlikely to happen in the near future.

Reply:
It is _always_ a good idea to remove hardcoded paths. With the rational "it's not going to change anyway" we could just as well get rid of find_directory in the OS.

Reply:
Like you said:
>I guess such a change is unlikely to happen in the near future.
So i don't think we need to change it _now_
Beside, see my comment on GCI

Reply:
The problem with that is, that you are touching it now, so it's a reasonably easy and low overhead change to do now. If the hardcoded path is kept now, noone will feel responsible and bother to change it later.  At the very least add a TODO so that it can be found more easily.

Also, hardcoding the python path like that is a bad idea, as that will break not only if the system path changes, but also when the user installs python in a different install location (i.e. into home). That is exactly what the `/package-links` directory with its automatic links was made for.

Reply:
Thanks @mmlr for your remark. I have to admit that you are right.

@tts2k, shall we keep this PR open until we reach a consensus?

Maybe we could even install the handler directly in `$settingsDir/profile.d/` (making it visible in `/boot/system/settings/profile.d/`) instead of installing a post-install script.

Reply:
@mmlr is right and i'm not going against him
But this recipe is gonna be rewritten sooner or later, because using post install script is definately not a good option for long-term.
@fbrosson just keep this PR for a few days, i'll try to rewrite it if i have free time

Reply:
Thanks @mmlr for suggesting the use of the package links. That's a very good idea.

@tts2k, these package links work this way:
When haikuporter sees a `cmd:python` is in REQUIRES, it creates a `cmd~python` symlink in a special directory which command_not_found will see. Then whenever a script (from that package) needs to access any file provided by the package which provides `cmd:python`, it just needs to look in `$portPackageLinksDir/cmd~python/`, which will be pointing to the `$prefix` of the package providing `cmd:python`.
This `portPackageLinksDir` variable is defined in a temporary file called `wrapper-script` which haikuporter creates in the `haikuports/app-shells/command_not_found/work-0.0.1~git` when it starts building that package.

And to really see these package links, you can call `haikuporter --enter-chroot command_not_found` (or `haikuporter -E command_not_found`) and then go to `/packages/command_not_found-0.0.1~git-1/ and you will see the `cmd~python` symlink to a directory. To exit the chroot you just need to type exit. Sometimes you need to type this twice.

This package links stuff is a bit complex to use, but this is one of the key features of the packaging.